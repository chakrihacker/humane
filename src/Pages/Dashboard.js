import React, {useEffect} from 'react'
import styled from "styled-components";
import Header from '../Components/Header';
import {  Route, Switch, useRouteMatch, useHistory } from "react-router-dom";
import SpendHistory from './SpendHistory';
import ClientContacts from './ClientContacts';

const StyledDashboard = styled.div`
  ${({ theme }) => `
		background-color: ${theme.palette.grey[900]};
		color: #fff;
		height: 100vh;
		display: flex;
		flex-direction: column;
	`}
`;

const Dashboard = () => {
	let { path } = useRouteMatch();
	let history = useHistory()

	useEffect(() => {
		let token = localStorage.getItem("token")
		if(!token) {
			history.push("/login")
		}
	// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [])

	return (
		<StyledDashboard>
			<Header />
			<Switch>
				<Route path={`${path}/spend-history`}>
					<SpendHistory />
				</Route>
				<Route path={`${path}/client-contacts`}>
					<ClientContacts />
				</Route>
			</Switch>
		</StyledDashboard>
	)
}

export default Dashboard
