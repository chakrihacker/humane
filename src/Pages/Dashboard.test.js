import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/react-testing";
import Dashboard from "./Dashboard";
import { ThemeProvider } from "styled-components";
import { createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme();

test("should match snapshot", () => {
  const {container} = render(
    <MockedProvider mocks={[]}>
      <ThemeProvider theme={theme}>
        <Dashboard />
      </ThemeProvider>
    </MockedProvider>,
    { wrapper: MemoryRouter }
	);
	expect(container).toMatchSnapshot()
});
