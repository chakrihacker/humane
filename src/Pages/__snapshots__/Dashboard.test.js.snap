// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="sc-AxirZ cfaoDL"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular sc-AxjAm fZVaJy MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Humane
        </h6>
        <div>
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</div>
`;
